File: src/main/java/fm/comunas/ddd/doc/model/DocPartTransition.java
package fm.comunas.ddd.doc.model;

import java.time.OffsetDateTime;

import fm.comunas.ddd.doc.model.enums.CodeCaseStage;
import fm.comunas.ddd.oe.model.ObjUser;

public interface DocPartTransition extends DocPart<Doc> {

	Integer getSeqNr();

	ObjUser getUser();

	OffsetDateTime getModifiedAt();

	CodeCaseStage getOldCaseStage();

	CodeCaseStage getNewCaseStage();

	String getChanges();

}

File: fm-dddrive/src/main/java/io/dddrive/doc/model/DocPartTransition.kt
package io.dddrive.doc.model

import io.dddrive.doc.model.enums.CodeCaseStage
import io.dddrive.oe.model.ObjUser
import java.time.OffsetDateTime

interface DocPartTransition : DocPart<Doc> {

	val seqNr: Int

	val user: ObjUser

	val timestamp: OffsetDateTime

	val oldCaseStage: CodeCaseStage?

	val newCaseStage: CodeCaseStage

	fun init(
		userId: Any,
		timestamp: OffsetDateTime,
		oldCaseStage: CodeCaseStage?,
		caseStage: CodeCaseStage,
	)

}
